SD-K9/src/app/components/building-info/building-info.component.ts

```diff
@@ -69,7 +71,10 @@ export class BuildingInfoComponent implements OnInit, OnDestroy {
 
   private _displayCard() {
     if (this.currentCampus !== '' && this.currentBuilding !== '') {
-      this.currentBuildingInfo = this._campusConfig[this.currentCampus]["buildings"][this.currentBuilding]['info'];
+      this.currentBuildingDepartmentsInfo = this._campusConfig[this.currentCampus]["buildings"][this.currentBuilding]['departments'];
+      this.currentBuildingServicesInfo = this._campusConfig[this.currentCampus]["buildings"][this.currentBuilding]['services'];
+      if(this.currentBuildingDepartmentsInfo == undefined) this.currentBuildingDepartmentsInfo = "no information to show";
+      if(this.currentBuildingServicesInfo == undefined) this.currentBuildingServicesInfo = "no information to show";
```

Comment:
Why is this necessary if you are defining the variables above it?

Reply:
Because this information is fetched from a JSON file and if the attributes dont exist in the object (were not specified in the file), they will be undefined.